File: docs/examples/python-project/services/order.py
# -*- coding: utf-8 -*-
"""订单服务：演示循环依赖 + 影响分析。"""
from services.inventory import check_stock, reserve


def create_order(items, user_id):
    """创建订单：检查库存 → 预留 → 返回订单号"""
    if not check_stock(items):
        raise ValueError("库存不足")
    return reserve(items, user_id)


def cancel_order(order_id):
    """取消订单：释放库存"""
    from services.inventory import release
    release(order_id)
    return f"order-{order_id}-cancelled"

File: docs/examples/js-project/services/order.js
// 订单服务：普通函数 + 箭头函数
export function createOrder(user, items) {
  const total = items.reduce((sum, it) => sum + priceOf(it), 0);
  return { user: user.id, items, total };
}

const priceOf = (item) => item.length * 10;

export function cancelOrder(orderId) {
  return `cancelled-${orderId}`;
}
